addons/binding/org.openhab.binding.wso2iots/src/main/java/org/openhab/binding/wso2iots/handler/wso2iotsHandler.java

```diff
@@ -0,0 +1,390 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+/**
+ * The {@link wso2iotsHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author wso2_Ramesha - Initial contribution
+ */
+// @NonNullByDefault
+public class wso2iotsHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(wso2iotsHandler.class);
+
+    private static final int DEFAULT_REFRESH_PERIOD = 60;
+
+    private ScheduledFuture<?> refreshJob;
+
+    private BigDecimal iotsResponseTemp;
+
+    private BigDecimal iotsResponseLight;
+
+    private BigDecimal iotsResponseMotion;
+
+    private BigDecimal iotsResponseHumidity;
+
+    private final Gson gson;
+
+    public wso2iotsHandler(Thing thing) {
+        super(thing);
+        gson = new Gson();
+    }
+
+    /*
+     * @Override
+     * public void handleCommand(ChannelUID channelUID, Command command) {
+     * if (channelUID.getId().equals(CHANNEL_1)) {
+     * // TODO: handle command
+     *
+     * // Note: if communication with thing fails for some reason,
+     * // indicate that by setting the status with detail information
+     * // updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+     * // "Could not control device at IP address x.x.x.x");
+     * }
+     * }
+     */
+    @Override
+    public void initialize() {
+        // TODO: Initialize the thing. If done set status to ONLINE to indicate proper working.
+        // Long running initialization should be done asynchronously in background.
+        // updateStatus(ThingStatus.ONLINE);
+
+        // Note: When initialization can NOT be done set the status with more details for further
+        // analysis. See also class ThingStatusDetail for all available status details.
+        // Add a description to give user information to understand why thing does not work
+        // as expected. E.g.
+        // updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+        // "Can not access device as username and/or password are invalid");
+
+        logger.debug("Initializing wso2iots handler.");
+
+        wso2iotsConfiguration config = getConfigAs(wso2iotsConfiguration.class);
+        logger.debug("config apikey = (omitted from logging)");
+        logger.debug("config deviceId = {}", config.deviceId);
+        logger.debug("config refresh = {}", config.refresh);
+
+        boolean validConfig = true;
+        String errorMsg = null;
+
+        if (StringUtils.trimToNull(config.apikey) == null) {
+            errorMsg = "Parameter 'apikey' is mandatory and must be configured";
+            validConfig = false;
+        }
+        if (StringUtils.trimToNull(config.deviceId) == null) {
+            errorMsg = "Parameter 'deviceId' is mandatory and must be configured";
+            validConfig = false;
+        }
+        if (config.refresh != null && config.refresh < 2) {
```

Comment:
no logging when refresh is null

Reply:
If refresh is null it takes DEFAULT_REFRESH_PERIOD as refresh period in the startAutomaticRefresh method

Reply:
but this should never occur and would mean some problem, the refresh period will always be passed on from the configuration, because it has a default value in the xml